### PR TITLE
Expose instance_aware flag in MSAL config

### PR DESCRIPTION
**Pull request by oldalton**

### CODE REVIEW

File: MSAL/src/public/configuration/MSALPublicClientApplicationConfig.h [hunk 1]
```diff
@@ -78,6 +78,14 @@ to target MSAL at a specific test slice & flight. These apply to all requests ma
 /** MSAL configuration interface responsible for token caching and keychain configuration. Refer to `MSALCacheConfig` for more details */
 @property (readonly) MSALCacheConfig *cacheConfig;
 
+/**
+ For clients that support multiple national clouds, set this to YES. NO by default.
+ If set to YES, the Microsoft identity platform will automatically redirect user to the correct national cloud during the authorization flow. You can determine the national cloud of the signed-in account by examining the authority associated with the MSALResult. Note that the MSALResult doesn't provide the national cloud-specific endpoint address of the resource for which you request a token.
+ 
+ @note Your client_id needs to be registered in national clouds for this feature to work.
+ */
+@property (nonatomic) BOOL multipleCloudsSupported;
```
Comment (antrix1989) [inline MSAL/src/public/configuration/MSALPublicClientApplicationConfig.h]: Thanks for not using instanceAware in public api. This name is much better :)

File: MSAL/src/configuration/MSALPublicClientApplicationConfig.m [hunk 2]
```diff
@@ -112,6 +112,7 @@ - (id)copyWithZone:(NSZone *)zone
     item->_cacheConfig = [_cacheConfig copyWithZone:zone];
     item->_verifiedRedirectUri = [_verifiedRedirectUri copyWithZone:zone];
     item->_extraQueryParameters = [_extraQueryParameters copyWithZone:zone];
+    item->_multipleCloudsSupported = _multipleCloudsSupported;
```
Comment (jasoncoolmax) [inline MSAL/src/configuration/MSALPublicClientApplicationConfig.m]: Seems we didn't propagate such a property down to the broker request construction.
We might want to propagate such flag to the broker request payload, either in the extraQueryParameters or as a separate parameter:
https://github.com/AzureAD/azure-activedirectory-tokenbroker-for-objc/blob/dev/ADAuthenticationBroker/ADAuthenticationBroker/ADBrokerRequest.m#L126

File: MSAL/src/public/configuration/MSALPublicClientApplicationConfig.h (hunk 1 above)
Comment (oldalton) [reply MSAL/src/public/configuration/MSALPublicClientApplicationConfig.h]: Name is courtesy of the Android team :)

File: MSAL/src/configuration/MSALPublicClientApplicationConfig.m (hunk 2 above)
Comment (oldalton) [reply MSAL/src/configuration/MSALPublicClientApplicationConfig.m]: Great catch! Will fix.

**Pull request merged**